### ISSUE TITLE
Allow job callables to receive Job instance as an argument

### ISSUE THREAD

**Issue by agronholm** [agronholm/apscheduler#89]
Originally reported by: **Ulrich Petri (Bitbucket: [ulope](http://bitbucket.org/ulope), GitHub: [ulope](http://github.com/ulope))**

---

Currently it's impossible (without resorting to hackery like `inspect.getframe()`) to access the `Job` instance from within the actual job callable.

I'd suggest to add an argument `pass_job` or similar to `add_job()` which if set to `True` would cause the job callable to be passed an additional keyword argument `job` that contains the Job instance.

---
- Bitbucket: https://bitbucket.org/agronholm/apscheduler/issue/89

**Comment by selimb** [comment]
Still interested in implementing this? In my case I'd like for the callable function to use a job-specific logger, e.g.

```
def scrape(url, job_id):
    logger = getLogger('myapp.' + job_id)
    # do stuff

scheduler = BackgroundScheduler()

scheduler.add_job(scrape, 'some_url', pass_job=True)
scheduler.add_job(scrape, 'some_url', pass_job=True)
```

The workaround would be to generate the IDs myself and pass it to the function, but why would I when APScheduler already does this.

**Comment by agronholm** [comment]
This has been implemented in v4.0.0a1 via the contextvar `apscheduler.current_job`.
